daal4py/mb/model_builders.py

```diff
@@ -283,8 +283,16 @@ def predict_proba(self, X):
 
 
 def convert_model(model):
-    gbm = GBTDAALModel()
-    gbm._convert_model(model)
+    try:
+        gbm = GBTDAALModel()
+        gbm._convert_model(model)
+    except TypeError as err:
+        if "Only GBTDAALRegressor can be created" in str(err):
+            gbm = d4p.sklearn.ensemble.GBTDAALRegressor.convert_model(model)
+        elif "Only GBTDAALClassifier can be created" in str(err):
+            gbm = d4p.sklearn.ensemble.GBTDAALClassifier.convert_model(model)
```

Comment:
Is it possible to check conditions for this kind of conversion avoiding exception?

Reply:
As discussed with @razdoburdin, this will be done in a rework of the dispatching logic. It's not the only thing we're doing suboptimally.